### PR TITLE
Handling empty outputs

### PR DESCRIPTION
**Pull request by natha7**

### CODE REVIEW

File: __tests__/endpoints.test.js [hunk 1]
```diff
@@ -137,7 +137,17 @@ describe("GET: /api/articles/:article_id/comments", () => {
...
+      });
+  });
+  test("GET 200: Returns an empty comments array when article id exists but the article has no associated comments", () => {
+    return request(app)
+      .get("/api/articles/10/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const comments = body.comments;
+        expect(Array.isArray(comments)).toBe(true);
+        expect(comments).toHaveLength(0);
```
Comment (nc-jo-watson) [inline __tests__/endpoints.test.js:150]: You could collapse these tests into a single test
`expect(comments).toEqual([])`

File: controllers/articles.controller.js [hunk 2]
```diff
@@ -28,9 +28,11 @@ exports.getArticles = (request, response, next) => {
 exports.getCommentsByArticleId = (request, response, next) => {
   const id = request.params.article_id;
 
-  fetchCommentsByArticleId(id)
-    .then((comments) => {
-      response.status(200).send({ comments });
+  const promises = [fetchCommentsByArticleId(id), fetchArticleById(id)];
```
Comment (nc-jo-watson) [inline controllers/articles.controller.js:31]: Nice!! 🌟

Comment (nc-jo-watson) [reply controllers/articles.controller.js:31]: bonus points if you can give this array a more descriptive name... 😄

**Review by nc-jo-watson — COMMENTED** [review]
Good solution Nathan! 👍 I've left picky comments on how to make your code cleaner 😄

File: controllers/articles.controller.js (hunk 2 above)
Comment (natha7) [reply controllers/articles.controller.js:31]: Thank you! should I include these changes in the next pr or push them before branching?

**Pull request merged**